core/code/chat.js

```diff
@@ -302,6 +308,30 @@ window.chat.nicknameClicked = function(event, nickname) {
 }
 
 window.chat.writeDataToHash = function(newData, storageHash, isPublicChannel, isOlderMsgs) {
+
+  if (newData.result.length > 0) {
+    //track oldest + newest timestamps/GUID
+    let first = {
+      guid: newData.result[0][0],
+      time: newData.result[0][1]
+    };
+    let last = {
+      guid: newData.result[newData.result.length-1][0],
+      time: newData.result[newData.result.length-1][1]
+    };
+    if (storageHash.oldestTimestamp === -1 || storageHash.oldestTimestamp >= last.time) {
+      if (isOlderMsgs || storageHash.oldestTimestamp != last.time) {
+        storageHash.oldestTimestamp = last.time;
+        storageHash.oldestGUID = last.guid;
+      }
+    }
+    if (storageHash.newestTimestamp === -1 || storageHash.newestTimestamp <= first.time) {
+      if (!isOlderMsgs || storageHash.newestTimestamp != first.time) {
+        storageHash.newestTimestamp = first.time;
+        storageHash.newestGUID = first.guid;
+      }
+    }
+  }
```

Comment:
I would recommend moving this to a new function for better code style.
`writeDataToHash` is already a bloated function.

Reply:
what about moving markup rendering to `chat.renderMsg` ?